core/shared/src/main/scala/fs2/text.scala

```diff
@@ -316,14 +316,34 @@ object text {
   def utf8EncodeC[F[_]]: Pipe[F, String, Chunk[Byte]] =
     utf8.encodeC
 
+  /** Transforms a stream of `String` such that each emitted `String` is a line from the input
+    * @param maxLineLength maximum size to accumulate a line to; throw an error if a line is larger
+    * @param crsOnly separate lines that are delimited only by '\r'
+    * @tparam F
+    */
+  def linesFor[F[_]: RaiseThrowable](
+      maxLineLength: Option[Int] = None,
+      crsOnly: Boolean = false
+  ): Pipe[F, String, String] =
+    linesImpl[F](
+      maxLineLength = maxLineLength.map((_, implicitly[RaiseThrowable[F]])),
+      crsOnly = crsOnly
+    )
+
   /** Transforms a stream of `String` such that each emitted `String` is a line from the input. */
-  def lines[F[_]]: Pipe[F, String, String] = {
+  def lines[F[_]]: Pipe[F, String, String] = linesImpl[F]()
+
+  private def linesImpl[F[_]](
+      maxLineLength: Option[(Int, RaiseThrowable[F])] = None,
+      crsOnly: Boolean = false
```

Comment:
I see, that `\r` can also be a line delimiter.
So, should it be a regression? i.e. should the default behavior change from treating `\n` & `\r\n` as a line delimiter to treating `\r`, `\r\n`, `\n` as a line delimiter? I would expect `lines` to handle this kind of ambiguity not with a flag, but always. In that case we should check that `\r\n` doesn’t produce two lines.

Reply:
My assumption is that other users wouldn't want the existing behavior to change. If I were the only consumer of this library I'd just have it always treat bare `\r`s as line separators.

I don't know about the consensus-forming process is here. But I'd be happy to go with the flow and do whatever other stakeholders agree on here.